pkg/plugin/aggregation/aggregator.go

```diff
@@ -100,6 +100,22 @@ func (a *Aggregator) isComplete() bool {
 	return true
 }
 
+// currentOrder returns the order of lowest order plugin which doesn't have a result yet. If all have results then -1 is returned.
+func (a *Aggregator) currentOrder() int {
```

Comment:
Hi @johnSchnake,

Please consider rewriting the function signature as `func (a Aggregator) currentOrder()` since there is no reason for the function to be able to write any values to the address of `a`. A read on a copy will suffice and offers better protection.

I do not see where the result Order is set. It seems to be related to the plug-in order. If plug-ins have a defined order, and the number of plug-ins is known, does that also mean the number of expected results are known? If so, then you could also maintain an analogue slice of addresses of the expected results that is sorted by order and use a `Search` function from the stdlib `sort` package. It may be overkill to do so though. I don't know whether or not the list is ever sufficiently large that keeping a second copy of addresses in memory is worth a faster search result.

Reply:
The list is usually very small; default is 2 and I could realistically imagine 5-10 being common for someone who is trying to run all their own custom, small plugins on each run. Even in a weird application I can't see it being an orders of magnitude higher to where we are concerned with the performance of a sort.